samples/wifi/provisioning/common/wifi_prov_handler.c

```diff
@@ -0,0 +1,566 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <zephyr/types.h>
+#include <zephyr/zephyr.h>
+#include <errno.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/wifi.h>
+#include <zephyr/net/wifi_mgmt.h>
+#include <zephyr/logging/log.h>
+
+#include <pb.h>
+#include <pb_encode.h>
+#include <pb_decode.h>
+
+#include "wifi_provisioning.h"
+#include "wifi_prov_internal.h"
+
+LOG_MODULE_REGISTER(wifi_prov, CONFIG_WIFI_PROVISIONING_LOG_LEVEL);
+
+#define WIFI_PROV_MGMT_EVENTS (NET_EVENT_WIFI_SCAN_RESULT | \
+				NET_EVENT_WIFI_CONNECT_RESULT)
+
+static struct net_mgmt_event_callback wifi_prov_mgmt_cb;
+
+/** Configurator should read info before any other operations.
```

Comment:
General note: it's a source file, not a header, no really need to use doxygen-style comments.